core/lib/types/src/commitment.rs

```diff
@@ -131,36 +131,40 @@ impl L1BatchWithMetadata {
     }
 
     pub fn l1_commit_data(&self) -> Token {
-        Token::Tuple(vec![
-            Token::Uint(U256::from(self.header.number.0)),
-            Token::Uint(U256::from(self.header.timestamp)),
-            Token::Uint(U256::from(self.metadata.rollup_last_leaf_index)),
-            Token::FixedBytes(self.metadata.merkle_root_hash.as_bytes().to_vec()),
-            Token::Uint(U256::from(self.header.l1_tx_count)),
-            Token::FixedBytes(self.metadata.l2_l1_merkle_root.as_bytes().to_vec()),
-            Token::FixedBytes(
-                self.header
-                    .priority_ops_onchain_data_hash()
-                    .as_bytes()
-                    .to_vec(),
-            ),
-            Token::Bytes(self.metadata.initial_writes_compressed.clone()),
-            Token::Bytes(self.metadata.repeated_writes_compressed.clone()),
-            Token::Bytes(self.metadata.l2_l1_messages_compressed.clone()),
-            Token::Array(
-                self.header
-                    .l2_to_l1_messages
-                    .iter()
-                    .map(|message| Token::Bytes(message.to_vec()))
-                    .collect(),
-            ),
-            Token::Array(
-                self.factory_deps
-                    .iter()
-                    .map(|bytecode| Token::Bytes(bytecode.to_vec()))
-                    .collect(),
-            ),
-        ])
+        if self.header.protocol_version.unwrap().is_pre_boojum() {
+            Token::Tuple(vec![
+                Token::Uint(U256::from(self.header.number.0)),
+                Token::Uint(U256::from(self.header.timestamp)),
+                Token::Uint(U256::from(self.metadata.rollup_last_leaf_index)),
+                Token::FixedBytes(self.metadata.merkle_root_hash.as_bytes().to_vec()),
+                Token::Uint(U256::from(self.header.l1_tx_count)),
+                Token::FixedBytes(self.metadata.l2_l1_merkle_root.as_bytes().to_vec()),
+                Token::FixedBytes(
+                    self.header
+                        .priority_ops_onchain_data_hash()
+                        .as_bytes()
+                        .to_vec(),
+                ),
+                Token::Bytes(self.metadata.initial_writes_compressed.clone()),
+                Token::Bytes(self.metadata.repeated_writes_compressed.clone()),
+                Token::Bytes(self.metadata.l2_l1_messages_compressed.clone()),
+                Token::Array(
+                    self.header
+                        .l2_to_l1_messages
+                        .iter()
+                        .map(|message| Token::Bytes(message.to_vec()))
+                        .collect(),
+                ),
+                Token::Array(
+                    self.factory_deps
+                        .iter()
+                        .map(|bytecode| Token::Bytes(bytecode.to_vec()))
+                        .collect(),
+                ),
+            ])
+        } else {
+            unimplemented!()
```

Comment:
I'll implement it after https://github.com/matter-labs/zksync-era/pull/259 is merged